### PR TITLE
[WIP] REVERT: "Outscale Builder: Get SSH Host from VM.Nics insead VM's Root"

### PR DESCRIPTION
**Pull request by gmlp**
This PR reverts   "Outscale Builder: Get SSH Host from VM.Nics insead VM's Root" https://github.com/hashicorp/packer/pull/8077 

closes: https://github.com/hashicorp/packer/issues/9108

### CODE REVIEW

**Comment by nywilken** [comment]
Hi @gmlp thanks for the contribution. I don't see an issue for this PR. Could you please open a [bug report,](https://github.com/hashicorp/packer/issues/new?labels=bug&template=bug_report.md) with the requested information, to accompany this PR. Or add the details in the PR's description if its easier. So that we may understand why you are proposing the change be reverted. Thanks!

**Comment by Hakujou** [comment]
Hello,

I created an issue here: https://github.com/hashicorp/packer/issues/9108

Thanks

File: builder/osc/common/step_run_source_vm.go [hunk 1]
```diff
@@ -243,7 +243,7 @@ func (s *StepRunSourceVm) Run(ctx context.Context, state multistep.StateBag) mul
 		}
 
 		if vm.PrivateIp != "" {
-			ui.Message(fmt.Sprintf("Private IP: %s", vm.PrivateIp))
+			ui.Message(fmt.Sprintf("Private IP: %s", vm.PublicIp))
```
Comment (SwampDragons) [inline builder/osc/common/step_run_source_vm.go:246]: Why this change?

File: builder/osc/common/omi_config.go [hunk 2]
```diff
@@ -25,6 +25,15 @@ type OMIConfig struct {
 	SnapshotGroups          []string `mapstructure:"snapshot_groups"`
 }
 
+func stringInSlice(s []string, searchstr string) bool {
```
Comment (SwampDragons) [inline builder/osc/common/omi_config.go:28]: This appears to be defined but unused. Why add it?
